app/helpers/templateGenerator.js

```diff
@@ -36,84 +58,136 @@ const getParamStrings = (component, params, type) => {
-    let modifiedValue;
-    if (component === 'view-data-bulletlist' && key === 'data') {
-      modifiedValue = (typeof value === 'string') ? value.split(',') : value;
-    } else {
-      modifiedValue = value.replace(opts.regex1, opts.string1).replace(opts.regex2, opts.string2);
-    }
-    const keyValueString = (i !== 0)
-      ? (`${opts.stringStart}${key}: ${JSON.stringify(modifiedValue)}`)
-      : (`${opts.stringStart0}${key}: ${JSON.stringify(modifiedValue)}`);
-    acc[0] += keyValueString;
-    return acc;
-  }, opts.accArray).join('');
+
+  const opts = options[isCode ? 'display' : 'code'];
+
+  let html = isCode ? '' : '<div class="nested-code">';
+
+  if (showKey) {
+    html += isCode ? '' : '<label for="param">';
+    html += '"';
+    html += key;
+    html += '": ';
+    html += isCode ? '' : '</label>';
+  }
+
+  if (isString) {
+    html += isCode ? '' : '<span  class="json-text-input">';
+    html += '"';
+    html += isCode ? '' : '<span contenteditable="true">';
+    html += newValue.replace(opts.regex1, opts.string1).replace(opts.regex2, opts.string2);
+    html += isCode ? '' : '</span>';
+    html += '"';
+    html += isCode ? '' : '</span>';
+    html += isLast ? ' ' : ',';
+    html += isCode ? '' : '<br>';
+  }
+
+  if (isObject) {
+    html += '{';
+    html += isCode ? '' : '<br><div class="json-object">';
+    html += Object.entries(newValue).map(([k, v], index) => generateEditableJSON([k, v, index + 1 === objectSize(newValue)], isCode)).join('');
+    html += isCode ? '' : '</div>';
+    html += '}';
+    html += isLast ? ' ' : ',';
+    html += isCode ? '' : '<br>';
+  }
+
+  if (isArray) {
+    html += '[';
+    html += isCode ? '' : '<br><div class="json-array">';
+    html += newValue.map((val, index) => generateEditableJSON([index || '0', val, index + 1 === objectSize(newValue), false], isCode)).join('');
+    html += isCode ? '' : '</div>';
+    html += ']';
+    html += isLast ? ' ' : ',';
+    html += isCode ? '' : '<br>';
+  }
+
+  html += isCode ? '' : '</div>';
+
+  return html;
+};
+
+const generateEditableCode = (params, isCode = false) => {
+  if (whatIsIt(params) === 'Array') {
+    return `[${params.map((val, index) => generateEditableJSON([index || '0', val, index + 1 === objectSize(params), false], isCode)).join('')}]`;
+  }
+  return `{${Object.entries(params).map((val, index) => generateEditableJSON([...val, index + 1 === objectSize(params)], isCode)).join('')}}`;
 };
 
-const getSettings = (component) => {
-  const settingsString = fs.readFileSync(`app/components/${component}/settings.json`, 'utf-8');
+/**
+ * Get the settings.json from the component/section directory.
+ * @param name
+ * @param type
+ * @returns {any}
+ */
+const getSettings = (name, type) => {
+  const settingsString = fs.readFileSync(`app/${type}s/${name}/settings.json`, 'utf-8');
   return JSON.parse(settingsString);
 };
 
+/**
+ * Write template to directory.
+ * @param template
+ * @param name
+ * @param type
+ */
```

Comment:
same here I think the name of the function and the signature don't need the comments